Add deprecation note about X.509/SHA-1 (#15581)
hashicorp/vault: website/content/docs/deprecation/faq.mdx

| Mount Filters         | [Path Filters](https://www.vaultproject.io/api-docs/system/replication/replication-performance#create-paths-filter) |
| AppID                 | [AppRole auth method](/docs/auth/approle)                                                                           |
| Standalone DB engines | [Combined DB engines](/docs/secrets/databases)                                                                      |

### Q: What is the impact of removing support for X.509 certificates with signatures that use SHA-1?

Starting with Vault 1.12.0, Vault will be built with Go 1.18.
The Go 1.18 standard library [rejects X.509 certificates](https://tip.golang.org/doc/go1.18#sha1) whose signatures use a SHA-1 hash.

If this issue impacts your usage of Vault, you can temporarily work around it by deploying Vault with the environment variable `GODEBUG=x509sha1=1` set.

> This workaround will fail in a future version of Go, but the Go team has not said when.

Reviewer: ```suggestion
This workaround will fail in a future version of Go, however, the Go team has yet to announce the Go version that will no longer require this workaround.
```

Reviewer: Something along those lines?

Reviewer 2: That's not strictly correct; at the point in time when this workaround is removed, the verification will fail entirely and the only way to work will be on an older Vault version. 

Maybe this is better?

```suggestion
This workaround will fail in a future version of Go, however, the Go team has not said when they will remove this workaround.
```

> Here are the use cases we that might use certificates with SHA-1 still:

Reviewer: ```suggestion
Here are the use cases that may still use certificates with SHA-1:
```

> - AWS Credential Plugin: AWS [still supports SHA-1 signatures](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html).

Author: I updated to clarify.
